### ISSUE TITLE
Next-generation mail application

### ISSUE THREAD

**Issue by Yoric** [Mozilla-Student-Projects/Projects-Tracker#60]
The objective of this project is to write an e-mail open web application that threats e-mails not as a list of messages but as a list of things to do.

How things are done is left open to the student(s) who take this project. The following is one possibility.

---

Whenever the user opens the application, they can choose between seeing:
1. the todo list (i.e. messages already read);
2. the new items, one by one (i.e. messages not read yet).

Choosing 1. shows a list of messages already read, sorted by deadline before action must be taken.

Choosing 2. shows the unread messages, one by one. There is no list. All messages are ready by order of arrival.

When the user reads a message, they have the option to:
- treat as unread -- moves the e-mail to the end of the list of unread e-mails;
- must handle today;
- must handle within 24h;
- must handle within 1 week;
- must handle within customizable time.
- already handled, archive;
- already handled, delete;
- already handled, mark as spam.

**Comment by hrishikeshs** [comment]
Hi, Where do I start?  I would like to try it.

**Comment by AntoineTurmel** [comment]
@Hrish2006 Look at the gaia email code:
https://github.com/mozilla-b2g/gaia/tree/master/apps/email

**Comment by jozefmorvay** [comment]
How do I get this assigned to me? I tried sending an email directly to Yoric's address, no response so far.

**Comment by AntoineTurmel** [comment]
@jozef300zx I can assign you on the project, @Hrish2006 would you like to work on it as well ?

Feel free to create a new repo and link it here.

**Comment by jozefmorvay** [comment]
@GeekShadow Can I ask you a few questions first? If you have an email address you won't mind sharing with me, I will send a couple of inquiries first, before taking on this project.

**Comment by aagamshah3** [comment]
@GeekShadow Hi, I would like to develop this app with two of my colleagues as our final year project. How do I get it assigned? And also, who would be mentoring us?

**Comment by AntoineTurmel** [comment]
@jozef300zx you can ask here if it's not really private else antoineturmel [at] gmail [dot] com

@aagamshah3 if you want to work along with @jozef300zx no problem

**Comment by jozefmorvay** [comment]
@GeekShadow No, it's not private. My question is more or less the same as aagamshah3's. While this project is supposed to be worked on independently and a certain level of knowledge is undoubtedly required , I would like to know if someone will provide a little guidance of what technologies to use, how to start etc.

**Comment by aagamshah3** [comment]
I've created a repo regarding this app https://github.com/aagamshah3/NGmail
I, along with my two colleagues, @rinimeledath @aadityasaraf would be contributing to this app.
We can surely work with @jozef300zx 
@GeekShadow We are beginners, so can we please know from where do we start? Who would be mentoring us?

**Comment by aagamshah3** [comment]
Please note the new repo: https://github.com/aagamshah3/Nextmail

**Comment by Yoric** [comment]
Assigned to @aagamshah3, @rinimeledath and @aadityasaraf

**Comment by jozefmorvay** [comment]
@Yoric would it be possible for two separate entities to work on this, seeing as I wanted this to be assigned to me before aagamshah3 and his team?

**Comment by Yoric** [comment]
@jozef300zx Certainly. There are many good ideas to be had in that domain. Both groups will need a mentor, preferably not the same one, though.

**Comment by jozefmorvay** [comment]
Agreed.

So then, would you be able to assign a mentor to me and the other group? I have never before been engaged in Mozilla student projects, so I do not know what the correct process is.

**Comment by jozefmorvay** [comment]
@Yoric I imagine you have so many more important things to do, and I don't want to bother you too much, but I would really appreciate a comment on this topic from you. I am looking forward to starting this project on my own, and would like to have a mentor assigned, who would guide me through next steps.

**Comment by Yoric** [comment]
Sorry for the radio silence. I do not assign mentors myself. Rather, I would prefer if you came over to chat with the Mozilla community and found a mentor. You can try on channels [openwebapps](http://client00.chat.mibbit.com/?server=irc.mozilla.org&channel=%23openwebapps&nick=jozef300zx) or [introduction](http://client00.chat.mibbit.com/?server=irc.mozilla.org&channel=%23introduction&nick=jozef300zx), for instance.

**Comment by jozefmorvay** [comment]
@Yoric I did as you advised. On the channel introduction, I was told to try and engage a would-be mentor through direct mozilla distro lists. I sent an email to these: dev-webapps@lists.mozilla.org, dev-webdev@lists.mozilla.org, webapps@lists.mozilla.org, now awaiting moderator approval. 

If there is yet another feasible way how to reach possible mentor, I would be happy to try it.

**Comment by DavidBruant** [comment]
Volunteering as mentor

**Comment by anistark** [comment]
hii, I would also like to work on it too. Iff it's still an ongoing project. Where do i start ?

**Comment by ghost** [comment]
Hi, I have the Gaia mail code and I wanna colaborate! Where do I start?

**Comment by aagamshah3** [comment]
@Yoric  @AntoineTurmel : We've made an initial commit. I request you to please have a look at it and give your feedback. We are also facing a couple of difficulties for which we need your guidance. How can we approach you? 
Link to the repo: https://github.com/aagamshah3/Nextmail
